File: FerramentaLayou/app/src/main/java/com/example/ferramentalayou/MainActivity.java
package com.example.ferramentalayou;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import android.widget.ToggleButton;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;
import androidx.core.view.ViewCompat;

public class MainActivity extends AppCompatActivity {
    //declarar as referencias do componente
    TextView txtTitulo; //obejto de texto disponivel
    EditText edtNomecompleto;

    SwitchCompat swMostrar;

    ToggleButton tbLigado;
    //Evento do botao
    Button btnConfirmar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //busca pelo id passando o id como parametro R de recuso, id é o que eu quero e o ultimo é o nome do id em si
        txtTitulo = findViewById(R.id.txtTitulo);
        txtTitulo.setText("Retrato");
        edtNomecompleto = findViewById(R.id.edtNomeCompleto);
        btnConfirmar = findViewById(R.id.btnConfirmar);
        swMostrar = findViewById(R.id.swMostrar);
        tbLigado = findViewById(R.id.tbLigado);

        //Evento para desabilitar botao
        tbLigado.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(tbLigado.isChecked())
                {
                    edtNomecompleto.setText("");
                    edtNomecompleto.setEnabled(false);
                }else{
                    edtNomecompleto.setEnabled(true);
                }
            }
        });

        //Evento para mudar algo com o switch
        swMostrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(swMostrar.isChecked())
                {
                    txtTitulo.setVisibility(View.GONE);
                }else{

                    txtTitulo.setVisibility(View.VISIBLE);
                    String titulo = txtTitulo.getText().toString().toUpperCase();
                    txtTitulo.setText(titulo);

                }
            }
        });


        //mensagem
        btnConfirmar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                if(TextUtils.isEmpty(edtNomecompleto.getText().toString())){
                    edtNomecompleto.setError("Você PRECISA DIGITAR ALGO");

                }else{
                    Toast.makeText(getBaseContext(), "Você digitou: "+edtNomecompleto.getText(), Toast.LENGTH_LONG).show();
                }




            }
        });


    }
}

File: ConversosModedasV2App/app/src/main/java/com/paulo/conversosmodedasv2app/MainActivity.kt
package com.paulo.conversosmodedasv2app

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.paulo.conversosmodedasv2app.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnDolar.setOnClickListener {
            calcular(0.8)
        }

        binding.btnEuro.setOnClickListener {
            calcular(5.9)
        }
        binding.btnPesor.setOnClickListener {
            calcular(1.2)
        }


    }

    private fun calcular(taxa: Double) {
        var reais = binding.edtEuros.text.toString().trim()
        if(!reais.isEmpty()){
            val resultado = reais.toDouble()*taxa;
            Toast.makeText(applicationContext, "${resultado}$", Toast.LENGTH_LONG).show();

        }

    }
}